smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/validation/SensitiveDataFinder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.typescript.codegen.validation;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.selector.Selector;
+import software.amazon.smithy.model.shapes.MapShape;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.traits.SensitiveTrait;
+import software.amazon.smithy.model.traits.StreamingTrait;
+
+/**
+ * This validator tells you whether a shape contains sensitive data fields.
+ * This is used to decide whether a sensitive log filter function needs to be generated for
+ * a given shape.
+ */
+public class SensitiveDataFinder {
+    private Map<Shape, Boolean> cache = new HashMap<>();
```

Comment:
How is this cache supposed to work? The public method takes in a model, which could change. I don't think this will work as expected, unless the same model is used every time.

Reply:
I changed the `model` to be a constructor parameter.